corehq/apps/users/models.py

```diff
@@ -2268,17 +2272,24 @@ def get_dimagi_emails_by_domain(cls, domain):
     def get_location_id(self, domain):
         return getattr(self.get_domain_membership(domain), 'location_id', None)
 
+    def _get_location_or_clear(self, domain, db_get_call):
+        loc_id = self.get_location_id(domain)
+        if loc_id:
+            try:
+                return db_get_call(loc_id)
+            except (ResourceNotFound, SQLLocation.DoesNotExist):
```

Comment:
(non-blocking) the fact that the exceptions you trap are coupled with what you pass into `db_get_call` make this feel a little leaky-abstractiony. i think is totally subjective whether the DRY-ness here is worth it. I might lean towards no, but that's just an opinion.